### PR TITLE
Change CSIDriver object version to v1beta1 to be compatible with k8s version

### PR DESCRIPTION
**Pull request by Jiawei0227**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
1.16 and 1.17 tests are failing because we are using CSIDriver version v1. It only changes to v1 on 1.18. So before that we should use v1beta1 for CSIDriver object.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```
/cc @verult 
/cc @msau42 
/cc @saikat-royc

### CODE REVIEW

**Comment by saikat-royc** [comment]
Should we include the changes of csidriver for prow-staging-head overlay too? @msau42

**Comment by msau42** [comment]
> Should we include the changes of csidriver for prow-staging-head overlay too? Michelle Au

Sure that's a good practice to follow

File: deploy/kubernetes/overlays/prow-gke-release-staging-head/csi-driver.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: storage.k8s.io/v1beta1
+kind: CSIDriver
+metadata:
+  name: pd.csi.storage.gke.io
+spec:
+  attachRequired: true
+  podInfoOnMount: false
```
Comment (saikat-royc) [inline deploy/kubernetes/overlays/prow-gke-release-staging-head/csi-driver.yaml:7]: nits:
the file complains "no newline at the end of file"

Comment (Jiawei0227) [reply deploy/kubernetes/overlays/prow-gke-release-staging-head/csi-driver.yaml]: Done

**Pull request merged**